### DISCUSSION TITLE
multi-bot team work

### DISCUSSION THREAD

**Discussion by donghao1393** [cline/cline#1038] [Feature Requests]
currently, there's only 1 bot accounting for coding, requirement managing (reading and monitoring), checking and others.
Can we split each role separately for different tasks in each high level / low level? it's like people in the team, which must involving project manager, architect, programmer, QA, etc so that attention would be more focused and more professional.
I believe it would make quality better and efficiency higher.

I was working on a project with cline. I made a task for it to fix a bug in page style. It spent ~$8 doing a lot of modification but nothing changed. I made the same chat in claude desktop and kept asking claude to check its finding thoroughly until it confirmed, i.e. I am the manager here. Then it solved the issue easily.

**Comment by d121122gma** [comment]
> currently, there's only 1 bot accounting for coding, requirement managing (reading and monitoring), checking and others.
> 
> Can we split each role separately for different tasks in each high level / low level? it's like people in the team, which must involving project manager, architect, programmer, QA, etc so that attention would be more focused and more professional.
> 
> I believe it would make quality better and efficiency higher.
> 
> 
> 
> I was working on a project with cline. I made a task for it to fix a bug in page style. It spent ~$8 doing a lot of modification but nothing changed. I made the same chat in claude desktop and kept asking claude to check its finding, i.e. I am the manager here. Then it solved the issue easily.

**Comment by rribeiro-ncode-pt** [comment]
Also possible, if it works, is the spawn of several subtasks generated by the llm itself (similar to tool use), like agenzero does, that whould make the context smaller/faster for each sub task, being the main task only responsible for planning and validating the results. This could probably reduce the costs and increase quality. (I would call it auto agentic behavior, optional would be even better, since some tasks don't need to be split and planed all that well)

**Reply by rribeiro-ncode-pt** [reply]
Note: Congrats to the author of Cline! I'm having a blast using it!
